classes/external.php

```diff
@@ -40,7 +41,8 @@ public static function dismiss_notice_parameters() {
     public static function dismiss_notice($noticeid) {
         $params = self::validate_parameters(self::dismiss_notice_parameters(),
             array('noticeid' => $noticeid));
-         return helper::dismiss_notice($params['noticeid']);
+        $notice = sitenotice::get_record(['id' => $params['noticeid']]);
```

Comment:
Will this explode if notice won't be found?

Reply:
It definitely will, fixing.

Reply:
Added some guard clauses for this one and `acknowledge_notice`